### PR TITLE
Add documentation for Lifecycle for Edge Network.

### PR DESCRIPTION
**Pull request by kevinlind**
Adds new section for Lifecycle for Edge Network.

- README with instructions how to add field group to schema, configure Lifecycle and Edge extensions in Launch, add rule in Launch to forward Lifecycle to Edge Network, and add Lifecycle and Edge to mobile application.
- API reference page
- Lifecycle for Edge Network event reference
- Lifecycle for Edge Network metrics reference.

### CODE REVIEW

File: foundation-extensions/lifecycle-for-edge-network/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Lifecycle for Edge Network
+
+The Adobe Experience Platform Lifecycle mobile extension enables application lifecycle data collection from your mobile app when using the Adobe Experience Platform Mobile SDK and the [Edge Network extension](../experience-platform-extension/).
+
+
```
Comment (praveek) [inline foundation-extensions/lifecycle-for-edge-network/README.md:5]: Can we add something like 
Follow these steps to forward lifecycle data to Platform.

File: foundation-extensions/lifecycle-for-edge-network/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+2. On the **Catalog** tab, locate or search for the **Mobile Core** extension, and click **Install**.
+3. There are no configuration settings for **Lifecycle for Edge Network**.
+4. Click **Save**.
+5. Follow the publishing process to update SDK configuration.
+
+In addition, use the following steps to [Configure the AEP Edge Network extension in Launch](https://aep-sdks.gitbook.io/docs/foundation-extensions/experience-platform-extension#configure-the-edge-network-extension-in-launch).
+
+![AEP Mobile Core extension configuration](../../.gitbook/assets/mobile-core-launch-configuration.png)
+
+## Configure Forward to Edge Rule in Launch
```
Comment (praveek) [inline foundation-extensions/lifecycle-for-edge-network/README.md:39]: Can we change this to Configure a Rule in Launch to forward Lifecycle metrics to Platform

File: foundation-extensions/mobile-core/lifecycle/lifecycle-event-reference.md [hunk 3]
```diff
@@ -72,3 +72,51 @@ Here are the key-value pairs in this event:
...
+
+### Lifecycle Application Foreground
+This event is a response from the Lifecycle extension to notify the application is being displayed in the foreground. The event is generated by the Lifecycle extension when the [lifecycleStart](https://aep-sdks.gitbook.io/docs/foundation-extensions/mobile-core/lifecycle/lifecycle-api-reference#lifecycle-start) API is called, typically when the application comes to the foreground of the device.
+
+{% hint style="info" %}
+To forward this event to the Adobe Experience Platform Edge Network, follow the steps outlined in [Configure Forward to Edge Rule in Launch](https://aep-sdks.gitbook.io/docs/foundation-extensions/lifecycle-for-edge-network#configure-forward-to-edge-rule-in-launch). 
+{% endhint %}
+
+{% hint style="info" %}
+In Android, there is a two (2) second timeout between consecutive `lifecycleStart` and `lifecyclePause` events to prevent dispatching of events due to switching between Android Activities.
```
Comment (praveek) [inline foundation-extensions/mobile-core/lifecycle/lifecycle-event-reference.md:86]: We changed it to 500 ms.

File: foundation-extensions/mobile-core/lifecycle/lifecycle-event-reference.md [hunk 4]
```diff
@@ -72,3 +72,51 @@ Here are the key-value pairs in this event:
...
+
+1. the `lifecyclePause` API is called, typically when the application goes to the background of the device.
+2. the `lifecycleStart` API is called without a previous call to `lifecyclePause` (such as when the application terminates unexpectedly). In this case, a Lifecycle Application Background event is dispatched just before the Lifecycle Application Foreground event, however the background event's timestamp is backdated to a time before the application previously went in the background.
+
+{% hint style="info" %}
+To forward this event to the Adobe Experience Platform Edge Network, follow the steps outlined in [Configure Forward to Edge Rule in Launch](https://aep-sdks.gitbook.io/docs/foundation-extensions/lifecycle-for-edge-network#configure-forward-to-edge-rule-in-launch). 
+{% endhint %}
+
+{% hint style="info" %}
+In Android, there is a two (2) second timeout between consecutive `lifecycleStart` and `lifecyclePause` events to prevent dispatching of events due to switching between Android Activities.
```
Comment (praveek) [inline foundation-extensions/mobile-core/lifecycle/lifecycle-event-reference.md:110]: 500 ms

**Review by stcheng-adobe — APPROVED** [review]

**Review by praveek — APPROVED** [review]

**Review by praveek — APPROVED** [review]

**Review by stcheng-adobe — APPROVED** [review]

**Review by stcheng-adobe — APPROVED** [review]

**Pull request merged**